rpc-client-api/src/filter.rs

```diff
@@ -170,6 +186,23 @@ impl Memcmp {
         }
     }
 
+    pub fn convert_to_raw_bytes(&mut self) {
```

Comment:
should this be fallible? same with optimize?  i'm struggling to reason about the behavior in the case of decode failure

Reply:
Ah, that's a good idea!
Current behavior is that `bytes_match()` always returns false if there is a decode failure, but might as well fail sooner (before we grab any accounts).

Reply:
Okay, I made `convert_to_raw_bytes` fallible, but just added a warning to optimize. Filter encoding should be verified with `verify_filter()` much earlier than optimize, so massaging different error types from optimize felt like more trouble than it was worth. Wdyt?